actors/common/src/main/java/org/sunbird/learner/actors/bulkupload/UserBulkUploadBackgroundJobActor.java

```diff
@@ -132,32 +133,56 @@ private void processUser(BulkUploadProcessTask task, String organisationId) {
             task, ProjectUtil.BulkProcessStatus.FAILED, ex.getMessage(), userMap, JsonKey.CREATE);
         return;
       }
-      if (userMap.get(JsonKey.ORG_ID) != null) {
-        orgMap = getOrg((String) userMap.get(JsonKey.ORG_ID));
-        if (orgMap == null) {
+      if (StringUtils.isNotBlank((String) userMap.get(JsonKey.ORG_ID))
+          || StringUtils.isNotBlank((String) userMap.get(JsonKey.EXTERNAL_ID))) {
+        orgMap = getOrgDetails(userMap);
+        if (MapUtils.isEmpty(orgMap)) {
           setTaskStatus(
               task,
               ProjectUtil.BulkProcessStatus.FAILED,
               ResponseCode.invalidOrgId.getErrorMessage(),
               userMap,
               JsonKey.CREATE);
           return;
-        }
-        if (!((String) orgMap.get(JsonKey.ROOT_ORG_ID)).equalsIgnoreCase(organisationId)) {
-          setTaskStatus(
-              task,
-              ProjectUtil.BulkProcessStatus.FAILED,
-              ResponseCode.invalidRootOrganisationId.getErrorMessage(),
-              userMap,
-              JsonKey.CREATE);
-          return;
+        } else {
+          if (StringUtils.isNotBlank((String) userMap.get(JsonKey.ORG_ID))
+              && StringUtils.isNotBlank((String) userMap.get(JsonKey.EXTERNAL_ID))) {
+            if (!((String) userMap.get(JsonKey.ORG_ID))
+                .equalsIgnoreCase((String) orgMap.get(JsonKey.ID))) {
+              String message =
+                  MessageFormat.format(
+                      ResponseCode.errorConflictingProperties.getErrorMessage(),
+                      JsonKey.ORGANISATION_ID,
+                      userMap.get(JsonKey.ORG_ID),
+                      JsonKey.EXTERNAL_ID,
+                      userMap.get(JsonKey.EXTERNAL_ID));
+              setTaskStatus(
+                  task, ProjectUtil.BulkProcessStatus.FAILED, message, userMap, JsonKey.CREATE);
+              return;
+            }
+          } else {
+            if (StringUtils.isNotBlank((String) userMap.get(JsonKey.EXTERNAL_ID))) {
+              userMap.put(JsonKey.ORGANISATION_ID, orgMap.get(JsonKey.ID));
+            } else {
+              userMap.put(JsonKey.ORGANISATION_ID, userMap.get(JsonKey.ORG_ID));
+            }
+          }
         }
       }
+      if (!((String) orgMap.get(JsonKey.ROOT_ORG_ID)).equalsIgnoreCase(organisationId)) {
+        setTaskStatus(
+            task,
+            ProjectUtil.BulkProcessStatus.FAILED,
+            ResponseCode.invalidRootOrganisationId.getErrorMessage(),
```

Comment:
Root organisation ID of API user is conflicting with that of specified organisation ID.